src/component.js

```diff
@@ -335,12 +346,13 @@ export default class Component {
         const observer = new MutationObserver((mutations) => {
             for (let i=0; i < mutations.length; i++) {
                 // Filter out mutations triggered from child components.
-                const closestParentComponent = mutations[i].target.closest('[x-data]')
+                const closestParentComponent = mutations[i].target.closest('[x-data], [data-x-data]')
 
                 if (! (closestParentComponent && closestParentComponent.isSameNode(this.$el))) continue
 
-                if (mutations[i].type === 'attributes' && mutations[i].attributeName === 'x-data') {
-                    const rawData = saferEval(mutations[i].target.getAttribute('x-data'), {})
+                if (mutations[i].type === 'attributes' && (mutations[i].attributeName === 'x-data' || mutations[i].attributeName === 'data-x-data')) {
```

Comment:
One way to clean this up would be 

```suggestion
                if (mutations[i].type === 'attributes' && ['x-data', 'data-x-data'].includes(mutations[i].attributeName)) {
```